eslint.config.mjs

```diff
@@ -0,0 +1,155 @@
+      'no-case-declarations': 'off',
+      'no-invalid-this': 'off',
+      'no-negated-condition': 'off',
+      'no-new': 'off',
+      'no-param-reassign': 'off',
+      'no-restricted-syntax': 'off',
+      radix: 'off',
+      'require-atomic-updates': 'off',
+      'jsdoc/match-description': [
+        'off',
+        { matchDescription: '^[A-Z`\\d_][\\s\\S]*[.?!`>)}]$' },
+      ],
+    },
+    settings: {
+      jsdoc: {
+        mode: 'typescript',
+      },
+    },
+  },
+  {
+    files: [
+      '**/jest.config.js',
+      '**/jest.environment.js',
+      '**/tests/**/*.{ts,js,tsx}',
+      '*.js',
+      '*.test.{ts,js,tsx}',
+      'scripts/*.ts',
+      'scripts/create-package/*.ts',
+      'yarn.config.cjs',
+    ],
+    extends: [nodejs],
+  },
+  {
+    files: ['*.test.{ts,js,tsx}', '**/tests/**/*.{ts,js,tsx}'],
+    extends: [jest],
+  },
+  {
+    // These files are test helpers, not tests. We still use the Jest ESLint
+    // config here to ensure that ESLint expects a test-like environment, but
+    // various rules meant just to apply to tests have been disabled.
+    files: ['**/tests/**/*.{ts,js,tsx}', '!*.test.{ts,js,tsx}'],
+    rules: {
+      'jest/no-export': 'off',
+      'jest/require-top-level-describe': 'off',
+      'jest/no-if': 'off',
+    },
+  },
+  {
+    files: ['*.js', '*.cjs'],
+    parserOptions: {
+      sourceType: 'script',
+      ecmaVersion: '2020',
+    },
+  },
+  {
+    files: ['*.ts', '*.tsx'],
+    extends: [typescript],
+    parserOptions: {
+      tsconfigRootDir: import.meta.dirname,
+      project: ['./tsconfig.packages.json'],
+    },
+    rules: {
+      // Enable rules that are disabled in `@metamask/eslint-config-typescript`
+      '@typescript-eslint/no-explicit-any': 'error',
+
+      // TODO: auto-fix breaks stuff
+      '@typescript-eslint/promise-function-async': 'off',
+
+      // TODO: re-enable most of these rules
+      '@typescript-eslint/no-unnecessary-type-assertion': 'off',
+      '@typescript-eslint/unbound-method': 'off',
+      '@typescript-eslint/prefer-enum-initializers': 'off',
+      '@typescript-eslint/prefer-nullish-coalescing': 'off',
+      '@typescript-eslint/prefer-optional-chain': 'off',
+      '@typescript-eslint/prefer-reduce-type-parameter': 'off',
+      'no-restricted-syntax': 'off',
+      'no-restricted-globals': 'off',
+    },
+  },
+  {
+    files: ['tests/setupAfterEnv/matchers.ts'],
+    parserOptions: {
+      sourceType: 'script',
+    },
+  },
+  {
+    files: ['*.d.ts'],
+    rules: {
+      '@typescript-eslint/naming-convention': 'warn',
+      'import/unambiguous': 'off',
+    },
+  },
+  {
+    files: ['scripts/*.ts'],
+    rules: {
+      // All scripts will have shebangs.
+      'n/shebang': 'off',
+    },
+  },
+  {
+    files: ['**/jest.environment.js'],
+    rules: {
+      // These files run under Node, and thus `require(...)` is expected.
+      'n/global-require': 'off',
+    },
+  },
+);
+
+export default config;
```

Comment:
ESLint configuration using the new [flat config format](https://eslint.org/docs/latest/use/configure/migration-guide)
Key differences from old .eslintrc format:
- Config is an array of objects, each acting like an "override" 
- Plugins/parsers are imported directly as modules
- Files patterns must be explicitly specified(we differentiate from [core(https://github.com/MetaMask/core/pull/4727/files#diff-9601a8f6c734c2001be34a2361f76946d19a39a709b5e8c624a2a5a0aade05f2) because we use `tsx` as well)
- All paths are relative to config file location
- No more "extends" - configs are composed using array spread